src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -380,6 +401,30 @@ async def common_validate(self, new, old, verrors):
                 "AD domain name is required."
             )
 
+    @private
+    async def config(self):
+        ha_mode = SMBHAMODE[(await self.middleware.call('smb.get_smb_ha_mode'))]
+        if ha_mode != SMBHAMODE.CLUSTERED:
+            return await self.middleware.call(
+                'datastore.config',
+                self._config.datastore,
+                {'extend': self._config.datastore_extend,
+                 'prefix': self._config.datastore_prefix}
+            )
```

Comment:
Its probably more future-proof if we use super() here

return await super().config()